extensions/xkcd/src/index.tsx

```diff
@@ -1,21 +1,46 @@
-import { ActionPanel, allLocalStorageItems, Color, Icon, List } from "@raycast/api";
+import { Action, ActionPanel, Color, Icon, List, LocalStorage } from "@raycast/api";
 import { useEffect, useState } from "react";
-import { maxNum } from "./xkcd";
-import { OpenComic, OpenRandomComic, OpenRandomUnreadComic } from "./open_comic";
+import { Comic, fetchComic, maxNum } from "./xkcd";
 import { useAtom } from "jotai";
-import { lastViewedAtom, maxNumAtom, readStatusAtom, totalReadAtom } from "./atoms";
+import { currentComicAtom, lastViewedAtom, maxNumAtom, readStatusAtom, totalReadAtom } from "./atoms";
+import getRandomUnread from "./get_random_unread";
+import OpenComicInBrowser from "./open_in_browser";
 
 export default function main() {
   const [num, setNum] = useAtom(maxNumAtom);
   const [readStatus, setReadStatus] = useAtom(readStatusAtom);
   const [loading, setLoading] = useState(true);
   const [totalRead] = useAtom(totalReadAtom);
   const [lastViewed, setLastViewed] = useAtom(lastViewedAtom);
+  const [comicData, setComicData] = useState<Comic | null>(null);
+  const [currentComic, setCurrentComic] = useAtom(currentComicAtom);
+  const [selectedId, setSelectedId] = useState("");
+  useEffect(() => {
+    if (currentComic === -1) return;
+    (async () => {
+      const data = await fetchComic(currentComic);
+      setComicData(data);
+      await LocalStorage.setItem(`read:comic:${currentComic}`, true);
+      await LocalStorage.setItem("last_viewed", currentComic);
+      setReadStatus({ ...readStatus, [currentComic]: true });
+      setLastViewed(currentComic);
+    })();
+  }, [currentComic]);
+  const markdownString =
+    comicData === null
+      ? null
+      : `
+# ${comicData.title} - #${comicData.num}
+
+![${comicData.alt}](${comicData.img})
+
+${comicData.alt}
```

Comment:
Do you need the `alt` text here? Because the image is loading async, it makes it jump in the detail view. It would be visually more pleasing if you put the image on the bottom and the text above. Or remove the `alt` text altogether. It should show up as tooltip when you hover the image.

Reply:
> It should show up as tooltip when you hover the image.

Hmm, this doesn't work for me, can you send a screenshot/recording of this behavior? I'll put the text above the image.

Reply:
Indeed it doesn't work... Though moving the text up makes it easier. Generally I'm not sure how much value the text brings. Is it for accessibility?

Reply:
The alt text is what you get when you mouseover the original xkcd comics, it generally adds some more commentary/a punchline to the comic